To help with #143 (LICENSE)

jeff-hykin opened the pull request:
Looks like I'm not the only one who would really like to use/extend this code. Any license would be appreciated, I just made a PR for MIT since its common for nix.

jeff-hykin committed:
Create LICENSE.md

guibou commented:
Thank you.

Is the licence forced to be "me" attributed (in that case, it is weird, but it should be `Guillaume Bouchard`, there as an e, even if my main email account does not have the `e`, that's maybe because there are something like a million 'Guillaume Bouchard" on the huge globe ;)

Otherwise, maybe we can just change the copyright to nix community?

Considering that I'm not maintaining nixGL and that I've transferred the account under `nix-community`.

I'll merge this asap once it this detail is clarified. beause looks like it is important, see https://github.com/orgs/nix-community/discussions/882#discussioncomment-10849248

soupglasses commented:
Following the nixpkgs standard, maybe something like:
`Copyright (c) Guillaume Bouchard and the nix-community/nixGL contributors`
would be good.

Its not the best but would follow what is the current precedent. 

Should put all current contributors as reviewers for the PR, so explicit permission to re-license the project to MIT can be done. There's a list of them at https://github.com/nix-community/nixGL/graphs/contributors

There's some information about how to re-license a project here: https://en.wikipedia.org/wiki/Software_relicensing. There is also such a thing as uncopyrightable changes, which is in handwavey terms is when they contain no creative expression, which may be helpful with some changes.

This is not legal advice, so do not rely on what i have said here as final or correct.

guibou commented:
Thank you @soupglasses.

Unfortunately, all of this is way above my mental capacity right now.

I'll document myself and try to find a bit of time to handle all of that.

zimbatm commented:
@dyfrgi if you open a new fixed PR without the .md, I will try to get in contact with Guillaume to get this merged.

jeff-hykin committed:
-

jeff-hykin commented:
> @dyfrgi if you open a new fixed PR without the .md, I will try to get in contact with Guillaume to get this merged.

If its any help, I removed the .md and corrected the name.

> @jeff-hykin Are you up for stewarding this PR? Looks like the next steps are to get explicit confirmation from the rest of the significant contributors.

I'm probably not active enough to track down every person, but I'll do what I can to help. I originally opened this thinking "oh the author just forgot a license".

zimbatm commented:
Thanks. Based on the previous conversation, I think we can assume that Guillaume is fine releasing his work under the MIT. The concern was more about how to go about it.

When it comes to licensing, the original author typically puts his name. It doesn't mean that other contributors give their copyright to him, it just sets the license type of the project.
